src/vsr/replica.zig

```diff
@@ -3457,18 +3463,28 @@ pub fn Replica(
             log.debug("{}: reset {} {s} message(s)", .{ self.replica, count, @tagName(command) });
         }
 
+        fn reset_quorum_counter(self: *Self, messages: *QuorumCounter) void {
+            var messages_iter = messages.iterator(.{});
+            while (messages_iter.next()) |replica| {
+                assert(replica < self.replica_count);
+            }
+
+            messages.setIntersection(QuorumCounterNull);
+            assert(messages.count() == 0);
```

Comment:
Great to see as this assert is critical. If the quorum is somehow not reset fully then we would have a correctness bug leading to data loss as quorums become overinflated without any real backing from real acks.

Reply:
Let's also add a loop up to `replica_count` asserting that all replica bits are unset, since this is so critical.